proposal.rst

```diff
@@ -239,7 +301,82 @@ Project Timeline
 .. any university classes or exams, vacations, etc), make sure you include them
 .. in your project timeline.
 
-FILLME
+MONTH 1  : 
+
+BIG TASK NAME : DOCUMENTATION STEP
+
+Week 1 
+	- Read and understand code-base further. 
+	- Create a new VagrantFile and make sure that it installs all the xapian dependencies. I have a doc where I have all commands which helped me with my Xapian setup.
+	- Update the "Installation" page with vagrant walk through tutorial.
+
+Week 2 
+	- Re-work on "Practical Example" tutorial to make it more user-friendly. 
+	- Add Omega documentation (in Xapian docs) for indexing, and how to obtain a front-end html(omindex, omega "query" [arguments])
+	- Start working on GSOC task:  "Rework Omega templates to use more modern web techniques"
+
+Week 3 
+	- Continue working on Omega templates.
+	- Identify areas that requires change user-end APIs, consult with mentors about this.
+	- Continue working on the user-end APIs.
+
+Week 4
+	- Continue working on the user-end APIs.
+	- Add tests to see consistency of new API(). 
+
+MONTH 1 DELVERABLE : 
+	[MINUMUM] : New "Practical Example" and "Installation" on Xapian Docs. Omega supporting templates (Subtask completed)	
+	[SATISFACTORY] : Above + New-API prototype + API-testing. Such that API is fully covered with unittests and use-cases.
+	
+
+MONTH 2  : 
+
+BIG TASK NAME : DOCUMENTATION STEP + REPORTING-SYSTEM
+DELEVARABLE OR SUBMITTABLE : 
+
+Week 1 
+	- Buffer week if things are not completed from previous month otherwise continue. 
+
+Week 2 
+	- Identify the requirements of a good reporting system(metrics, benchmarks, which algorithms etc..) as consulted by the developers on IRC and mentors
```

Comment:
Do you have an idea of which metrics and algorithms are relevant here? I'm not entirely clear from your proposal what this would look like. (Also, what are your measuring? Result relevance? Clustering? Impact of letor? All of the above?)